modules/explorer-core/src/main/scala/org/ergoplatform/explorer/db/repositories/DexOrdersRepo.scala

```diff
@@ -0,0 +1,157 @@
+import sigmastate.{SLong, Values}
+
+import scala.util.{Failure, Success, Try}
+
+/** [[ExtendedOutput]] for DEX sell/buy orders data access operations.
+  */
+trait DexOrdersRepo[D[_], S[_[_], _]] {
+
+  /** Get all unspent main-chain DEX sell orders
+    */
+  def getAllMainUnspentSellOrderByTokenId(
+    tokenId: TokenId
+  ): S[D, DexSellOrderOutput]
+
+  /** Get all unspent main-chain DEX buy orders
+    */
+  def getAllMainUnspentBuyOrderByTokenId(
+    tokenId: TokenId
+  ): S[D, DexBuyOrderOutput]
+}
+
+object DexOrdersRepo {
+
+  def apply[D[_]: LiftConnectionIO]: DexOrdersRepo[D, Stream] =
+    new Live[D]
+
+  final private class Live[D[_]: LiftConnectionIO] extends DexOrdersRepo[D, Stream] {
+
+    import org.ergoplatform.explorer.db.queries.{DexOrdersQuerySet => QS}
+
+    private val liftK = LiftConnectionIO[D].liftConnectionIOK
+
+    override def getAllMainUnspentSellOrderByTokenId(
+      tokenId: TokenId
+    ): Stream[D, DexSellOrderOutput] =
+      QS.getMainUnspentSellOrderByTokenId(
+          tokenId,
+          sellContractTemplate,
+          0,
+          Int.MaxValue
+        )
+        .stream
+        .map(eOut =>
+          DexSellOrderOutput(
+            eOut,
+            getTokenPriceFromSellOrderTree(eOut.output.ergoTree).get
+          )
+        )
+        .translate(liftK)
+
+    /** Get all unspent main-chain DEX buy orders
+      */
+    override def getAllMainUnspentBuyOrderByTokenId(
+      tokenId: TokenId
+    ): Stream[D, DexBuyOrderOutput] =
+      QS.getMainUnspentBuyOrderByTokenId(tokenId, buyContractTemplate, 0, Int.MaxValue)
+        .stream
+        .map(eOut =>
+          DexBuyOrderOutput(eOut, getTokenInfoFromBuyOrderTree(eOut.output.ergoTree).get)
+        )
+        .translate(liftK)
+
+  }
+
+  /** template for the buyer contract as of http://github.com/ScorexFoundation/sigmastate-interpreter/blob/42e55cbfd093252b8005e4607970764dd6610cbe/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L33-L45
+    */
+  val sellContractTemplate: HexString = HexString
+    .fromString[Try](
+      "eb027300d1eded91b1a57301e6c6b2a5730200040ed801d60193e4c6b2a5730300040ec5a7eded92c1b2a57304007305720193c2b2a5730600d07307"
+    )
+    .get
+
+  /** template for the buyer contract as of http://github.com/ScorexFoundation/sigmastate-interpreter/blob/42e55cbfd093252b8005e4607970764dd6610cbe/contract-verification/src/main/scala/sigmastate/verification/contract/AssetsAtomicExchange.scala#L12-L32
+    */
+  val buyContractTemplate: HexString = HexString
+    .fromString[Try](
+      "eb027300d1eded91b1a57301e6c6b2a5730200040ed803d601e4c6b2a5730300020c4d0ed602eded91b172017304938cb27201730500017306928cb27201730700027308d60393e4c6b2a5730900040ec5a7eded720293c2b2a5730a00d0730b7203"
+    )
+    .get
+
+  private val treeSerializer: ErgoTreeSerializer = new ErgoTreeSerializer
+
+  def getTokenPriceFromSellOrderTree(ergoTree: HexString): Try[Long] =
```

Comment:
Please move this static helper functions into some utils module

Reply:
Done.